0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+---
+
+# Schema
+
+TypeDB schema is like a blueprint of a database.
+
+TypeDB schema defines all user-defined types and rules used in a database.
+
+All user-defined types must subtype the entity, relation, or attribute base type or another user-defined type.
+
+As a result, TypeDB schemas look like ER diagrams. They resemble logical data models,
+removing the need to create a separate physical data model and allowing developers to
+create schemas that mirror their object model simplifying both design and development.
+
+![Types hierarchy](../../images/schema/thing-era-role-rule.png)
+
+TypeDB schemas are written with TypeQL which is both Data Definition Language and Data Manipulation Language (used 
+for schema and data queries). You can see a complete set of TypeQL syntax and grammar rules in ANTLR v.4 specification 
+file: [TypeQL.g4](https://github.com/vaticle/typeql/blob/master/grammar/TypeQL.g4).
+
+## Types
+
+Types define the data model, its constraints, and the relationships within it. In addition, the types within a schema 
+reflect the vocabulary used to describe the business domain it represents.
+
+Creating a type means subtyping a base type or a user-defined type. The base types are:
+
+|      Type      |              Instance of Type               |
+|:--------------:|:-------------------------------------------:|
+|  Entity type   |    Entity <br/>(Instance of Entity type)    |
+| Relation type  |  Relation <br/>(Instance of Relation type)  |
+| Attribute type | Attribute <br/>(Instance of Attribute type) |
+
+There is a strict hierarchy of types with all types being children (grandchildren, grand- grandchildren, etc.) 
+of a base type. Every type can have only one parent. There is also an internal type `role` that is used in relations. 
+Instances of types (i.e., the data) are known as `entities`, `relations`, and `attributes`.
+
+### Thing type
+
+There is a base type called `thing` type that can be used to address all types (both base and user-defined).
+
+<div class="note">
+[Warning]
+The `thing` base type will be deprecated in TypeDB version 3.0.
+</div>
+
+### Entity types
+
+Entity types define the objects within the business domain (e.g., an organization, file, or user). Further, an entity 
+(as in instance of an entity type e.g., Vaticle LTD) can exist independently of any other instance of any type 
+(e.g., John Doe).
+
+### Relation types
+
+Relation types define the roles participants can **play** in a relationship. The participants can be `entities`, 
+`relations`, or `attributes`. A `relation` type must specify at least one role. For example, `group-membership` 
+is a `relation` type that defines `user-group` and `group-member` roles. The `user-group` role is to be played 
+by a `user-group` entity whereas the `group-member` role is to be played by a `subject` entity.
+
+Roles allow the schema to enforce logical constraints. For example, a `group-membership` relation cannot associate 
```

Comment:
Stopping at "logical constraints" is a bit vague.
Everything typeql statement is a logical constraint.

Ideally, I would expect logical constraints to be in the domain of logical inference - statements such as 
`not { $x is a person; (shaver: $x, shaved: $x) isa shaving; };`

Reply:
But roles can enforce logical constraints. It's not the only way to enforce logical constraints but rather one of the ways.
We still need to talk about roles in this section. And why else do we need roles if not to limit what types can plays them?

Reply:
All of that is fine. It's just stopping the sentence at "logical constraints" sounds a bit too general. Maybe elaborating helps: 
> "Roles allow the schema to enforce logical constraints on which types can play them" ?